Doc/tools/extensions/peg_highlight.py

```diff
@@ -0,0 +1,79 @@
+from pygments.lexer import RegexLexer, bygroups, include
+from pygments.token import Comment, Generic, Keyword, Name, Operator, Punctuation, Text
+
+from sphinx.highlighting import lexers
+
+
+class PEGLexer(RegexLexer):
+    """Pygments Lexer for PEG grammar (.gram) files
+
+    This lexer strips the following elements from the grammar:
+
+        - Meta-tags
+        - Variable assignments
+        - Actions
+        - Lookaheads
+        - Rule types
+        - Rule options
+        - Rules named `invalid_*` or `incorrect_*`
+    """
+
+    name = "PEG"
+    aliases = ["peg"]
+    filenames = ["*.gram"]
+    _name = r"([^\W\d]\w*)"
+    _text_ws = r"(\s*)"
+
+    tokens = {
+        "ws": [(r"\n", Text), (r"\s+", Text), (r"#.*$", Comment.Singleline),],
+        "lookaheads": [
```

Comment:
I think we discussed this previously, but the only lookahead we should ignore are the ones that start a rule (the ones that we use for performance), the others I think we want to keep

Reply:
Yup, I was just about to say this. I think I'd prefer having to answer the question "Why are these lookaheads here?" rather than "Why is this grammar not working properly?".

> but the only lookahead we should ignore are the ones that start a rule

And we could limit this even further to only the *positive* lookaheads that start a rule.

Reply:
Done.